tests/unit/s2n_x509_validator_test.c

```diff
@@ -2188,5 +2189,130 @@ int main(int argc, char **argv)
         }
     }
 
+    /* Ensure that CA's outside of cert preferences aren't used */
```

Comment:
Nit: it took me a minute to figure out this means that we should fail for CAs that don't match our preferences. Maybe "Test that CAs must comply with cert preferences"?